lib/dqt/client.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module Dqt
+  class Client
+    def initialize(
+      headers: Dqt.configuration.client.headers,
+      host: Dqt.configuration.client.host,
+      params: Dqt.configuration.client.params
+    )
+      self.headers = headers
+      self.host = host
+      self.params = params
+    end
+
+    def api
+      @api ||= Api.new(client: self)
+    end
+
+    def get(path: "/", params: {})
+      http_client = HTTPClient.new
+
+      headers = {
+        'Content-Type': "application/json",
+      }.merge(send(:headers))
+
+      params = params.merge(send(:params))
+
+      response = Response.new(
+        response: http_client.get(
+          url(path),
+          header: headers,
+          query: params,
+        ),
+      )
+      raise ResponseError.new("DQT request failed", response) if [*0..199, *300..599].include? response.code
```

Comment:
3xx can potentially not be a failure, at least in http world, right?

Reply:
For this client, it's a failure. No data is a fail